inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+      mutate(AlnLength = as.numeric(AlnLength)) %>%
+      mutate(PcPositive = PcPosOrig)
+
+  } else if (wblast == FALSE) {
+    # If using classic-blast output
+    blast_out <- read_tsv(file = infile_blast, col_names = cl_blast_colnames)
+    cleanedup_blast <- blast_out %>%
+      mutate(AccNum = gsub("\\|", "", AccNum)) %>%
+      mutate(AccNum = gsub(".*[a-z]", "", AccNum)) %>%
+      mutate(Species = gsub(";.*$", "", Species)) %>%
+      mutate(PcIdentity = round(PcIdentity, 2)) %>%
+      mutate(PcPositive = round((PcPosOrig * AlnLength / QLength), digits = 2))
+
+    # Merge cleaned blast output with acc2info
+    cleanedup_blast <- merge(cleanedup_blast, acc2info_out,
+                             by.x = "AccNum",
+                             by.y = "FullAccNum",
+                             all.x = TRUE) %>%
+      select(-Species.x, -TaxID.x)
+    names(cleanedup_blast)[names(cleanedup_blast) == "Species.y"] <- "Species"
+  }
+
+  # TaxID to lineage mapping
+  cleanedup_blast$TaxID <- as.integer(cleanedup_blast$TaxID)
+  lineage_map <- fread("~/awasyn/new_trial/lineage_lookup.txt",
+                       header = TRUE, fill = TRUE,
+                       colClasses = lineage_map_cols)
+
+  # Merge with lineage map and clean up columns
+  mergedLins <- merge(cleanedup_blast, lineage_map, by = "TaxID",
+                      all.x = TRUE) %>%
+    mutate(Species = Species.y, Spp.blast = Species.x) %>%
+    select(any_of(cl_blast_postcln_cols))
+
+  # Add names and prepare the output
+  blast_names <- addName(mergedLins)
+
+  # Create output file name
+  file_name <- file.path(outdir, paste0(prefix, ".blast.cln.tsv"))
+
+  # Write cleaned data to file
+  write_tsv(blast_names, file_name, col_names = TRUE)
+}
+
+
+# Function to run BLASTCLUST on given input
+run_blastclust <- function(infile, suffix, outdir) {
+
+  # Prepare output file path
+  outfile <- file.path(outdir, paste0(suffix, ".bclust.L60S80.tsv"))
+  input_file <- file.path(outdir, paste0(suffix, ".bclust.L60S80.tsv.clstr"))
+
+
+  # Print process messages
+  cat("\n#####################################\n")
+  cat("## Now running BLASTCLUST on file(s):", infile, "\n")
+  cat("#####################################\n")
+
+  # Run BLASTCLUST
+  # blastclust_cmd <- paste("blastclust -i", infile, "-o", outfile, "-p T -L .6 -b T -S 80 -a 8")
+  cdhit_command <- sprintf(
+    "cd-hit -i %s -o %s -c 0.8 -aS 0.6 -T 8",
+    infile, outfile
+  )
+  clean_command <- sprintf(
+    "awk '/^>Cluster/ {if(NR>1)printf \"\\n\"; next} /WP_/ {start=index($0, \"WP_\"); if(start) {end=index(substr($0, start), \"...\"); if (end == 0) end=length($0); printf \"%%s \", substr($0, start, end-1)}}' %s > %s",
+    input_file,
+    outfile
+  )
+
+  cat("\nPerforming BLASTCLUST analysis on", infile, "\n")
+  # system(blastclust_cmd)
+
+  system(cdhit_command)
+  system(clean_command)
+
+}
+
+# Function to format blastclust output
+clust2tbl <- function(clust, blast) {
```

Comment:
Does this work equally well for blastclust & cdhit outputs? Tested fully?
@awasyn @epbrenner

Reply:
Also, I am guessing these functions exist in other places, too. Same Q as before to minimally repost functions that exist in `R`. Best to use this space ONLY for new functions.